src/codegen/compile.ml

```diff
@@ -1516,8 +1516,8 @@ module Word64 = struct
     let name = prim_fun_name Type.Nat64 "wpow_nat" in
     Func.share_code2 env name (("n", I64Type), ("exp", I64Type)) [I64Type]
       (fun env get_n get_exp ->
-        let set_n = G.i (LocalSet (nr 0l)) in
-        let set_exp = G.i (LocalSet (nr 1l)) in
+        let set_n = G.setter_for get_n in
```

Comment:
Is it really worth doing this rather than just creating a new local? I would (perhaps naively) expect the jit to optimize the local away and identify with the parameter.

If being able to assign to parameters is generally useful, I wonder if we should adjust the arguments to shared_codeX to be setter/getter pairs in general.

Reply:
Ah, I see that .setter_for also cleans up some existing code (serialization below); got for it.

Reply:
Exactly, I prefer having immutable parameters, but here I see justification to mutate. The extra local would inflate instruction count and probably cycles balances too, since instrumenting counts instructions too.